tests/lib/prepare.sh

```diff
@@ -843,12 +843,25 @@ setup_reflash_magic() {
     UNPACK_DIR="/tmp/$core_name-snap"
     unsquashfs -no-progress -d "$UNPACK_DIR" /var/lib/snapd/snaps/${core_name}_*.snap
 
-    # install ubuntu-image
-    snap install --classic --channel="$UBUNTU_IMAGE_SNAP_CHANNEL" ubuntu-image
+    if os.query is-core16; then
+        # the new ubuntu-image expects mkfs to support -d option, which was not
+        # supported yet by the version of mkfs that shipped with Ubuntu 16.04
+        snap install ubuntu-image --channel="$UBUNTU_IMAGE_SNAP_CHANNEL" --classic
```

Comment:
maybe we should define (and export) the `UBUNTU_IMAGE` variable at this point, since we currently define it in two places?

Reply:
It would have to be a separate helper in one of the scripts, just exporting it here would not make it availble in other shell fragments. OTOH maybe we could pull a trick and symlink to the right one, but that sounds like a followup material anyway.